backend/api/tests/views/test_views_compute_plan_graph.py

```diff
@@ -51,25 +50,78 @@ def test_too_many_tasks(self):
 
     def test_cp_graph(self):
         compute_plan = factory.create_computeplan()
-        algo = factory.create_algo()
+
+        algo_data_to_model = factory.create_algo(
+            inputs=factory.build_algo_inputs(["datasamples", "opener"]),
+            outputs=factory.build_algo_outputs(["model"]),
+        )
+
+        algo_model_to_model = factory.create_algo(
+            inputs=factory.build_algo_inputs(["model"]),
+            outputs=factory.build_algo_outputs(["model"]),
+        )
+
+        algo_model_to_performance = factory.create_algo(
+            inputs=factory.build_algo_inputs(["model"]),
+            outputs=factory.build_algo_outputs(["performance"]),
+        )
+
+        algo_multiple_model_to_model = factory.create_algo(
+            inputs=factory.build_algo_inputs(["model"]),
+            outputs=factory.build_algo_outputs(["model"]),
+        )
+
         train_task = factory.create_computetask(
             compute_plan,
-            algo=algo,
-            category=ComputeTask.Category.TASK_TRAIN,
+            rank=1,
+            algo=algo_data_to_model,
+            outputs=factory.build_computetask_outputs(algo_data_to_model),
         )
+
         predict_task = factory.create_computetask(
-            compute_plan, algo=algo, category=ComputeTask.Category.TASK_PREDICT, parent_tasks=[train_task.key]
+            compute_plan,
+            rank=2,
+            algo=algo_model_to_model,
```

Comment:
As you named the task `predict_task`, maybe you can rename the algo accordingly `predict_algo`.

Reply:
Same for the other algos.

Reply:
But I feel you want to output a prediction, not a model. You can find extensive examples in https://github.com/Substra/substra-backend/blob/main/backend/api/management/commands/generate_fixtures.py

Reply:
Actually we wanted to test the different kind of inputs/outputs, that's why the algo have those names. Maybe the task names are misleading and stick too much to the pre-generic task era. I agree that it would probably be predictions but if I'm not mistaken predictions is just an identifier and underlying it's a kind model I didn't see the point to make the difference in the context of this test. Having in mind that today we use meaningful names for those identifiers but it's meant to be free text.

Reply:
Having neat test data is better for human reading but obviously it's a minor point, feel free to ignore it.

Reply:
You're right! Test data improved in dc68ec8c541ea4f3cecbd94b4a19fc9ca825850a